src/engine/parser/implements/dml/select.rs

```
@@ -123,42 +123,63 @@ impl Parser {
+                                    "unexpected token '{:?}'",
+                                    current_token
+                                )));
+                            }
                         }
                     }
                 }
             }
         }
 
+        // GROUP BY ALL: expand to all non-aggregate columns
+        let needs_expand = query_builder.group_by_clause.as_ref().map_or(false, |c| c.group_by_all);
```

Clippy (1.96.0)

[warning] this `map_or` can be simplified

(warning)